### PR TITLE
#241

### PR DESCRIPTION
**Pull request by sergsever**
#241

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -25,15 +40,21 @@ ifdef DEBUG
 	EXTRA_CFLAGS += -DDEBUG=$(DEBUG)
 endif
 
+
```
Comment (krizhanovsky) [inline Makefile]: Extra empty line

File: Makefile [hunk 2]
```diff
@@ -17,6 +17,21 @@
 # this program; if not, write to the Free Software Foundation, Inc., 59
 # Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 
+uname_m = $(shell uname -m)
+ARCH = $(uname_m)
+ifeq ($(ARCH),x86_64)
+else
+ERROR="architecture must be x86_64 pressent:$(ARCH)\n"
```
Comment (krizhanovsky) [inline Makefile]: "\n" isn't treated by Make, so please remove it hereafter

File: Makefile [hunk 3]
```diff
@@ -17,6 +17,21 @@
 # this program; if not, write to the Free Software Foundation, Inc., 59
 # Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 
+uname_m = $(shell uname -m)
+ARCH = $(uname_m)
+ifeq ($(ARCH),x86_64)
+else
```
Comment (krizhanovsky) [inline Makefile]: It's better to replace ifeq-else with empty IF branch by ifneq,

        ifneq ($(ARCH),x86_64)
                ERROR="architecture must be x86_64 pressent:$(ARCH)\n"
       endif

Also please use preceding tab in the branch body as in the example above

File: Makefile [hunk 4]
```diff
@@ -17,6 +17,21 @@
...
+ERROR="architecture must be x86_64 pressent:$(ARCH)\n"
+endif
+PROC=$(shell cat /proc/cpuinfo)
```
Comment (krizhanovsky) [inline Makefile]: Please place ARCH and PROC definitions nearly, before the IF checks

File: Makefile [hunk 5]
```diff
@@ -25,15 +40,21 @@ ifdef DEBUG
...
 endif
 
+
 obj-m	+= tempesta_db/core/ tempesta_fw/
 
 KERNEL = /lib/modules/$(shell uname -r)/build
 
 export KERNEL EXTRA_CFLAGS
+uname_m = $(shell uname -m)
+ARCH = $(uname_m)
```
Comment (krizhanovsky) [inline Makefile]: Bot the variables are already declared above, why do you rediclare them?

File: Makefile [hunk 6]
```diff
@@ -17,6 +17,21 @@
 # this program; if not, write to the Free Software Foundation, Inc., 59
 # Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 
+uname_m = $(shell uname -m)
+ARCH = $(uname_m)
```
Comment (krizhanovsky) [inline Makefile]: uname_m is used only once, so it's not necessary

File: Makefile [hunk 7]
```diff
@@ -17,6 +17,21 @@
...
+ifeq ($(ARCH),x86_64)
```
Comment (krizhanovsky) [inline Makefile]: Please use spaces after commas

File: Makefile [hunk 8]
```diff
@@ -17,6 +17,21 @@
...
+else
+ERROR="architecture must be x86_64 pressent:$(ARCH)\n"
+endif
+PROC=$(shell cat /proc/cpuinfo)
+ifneq ("",$(findstring sse4_2,$(PROC))) 
+else
+ERROR="the sse4_2 support not found\n"
```
Comment (krizhanovsky) [inline Makefile]: The message is not infomative, the better is "SSE 4.2 support is required"

File: Makefile [hunk 9]
```diff
@@ -17,6 +17,21 @@
...
+endif
+ifneq ("",$(findstring pse,$(PROC))) 
+else
+ERROR="the pse support not found\n"
```
Comment (krizhanovsky) [inline Makefile]: Bad message, the better is "1MB huge pages support is required"

File: Makefile (hunk 1 above)
Comment (sergsever) [inline Makefile]: I'll fix it and include in next commits.
20 Окт 2015 г. 19:21 пользователь "Alexander Krizhanovsky" <
notifications@github.com> написал:

**Comment by krizhanovsky** [comment]
Already merged

**Pull request closed without merging**